### PR TITLE
feat: hexadecimal color literals in Style

### PR DESCRIPTION
**Pull request by wodeni**
# Description

Related issue/PR: #1094 

The non-standard `rgba` format Style used had been a pain to work with. This PR adds native support for hexadecimal color literals as specified in [web standards](https://developer.mozilla.org/en-US/docs/Web/CSS/hex-color), i.e.:

```
#RGB        // The three-value syntax
#RGBA       // The four-value syntax
#RRGGBB     // The six-value syntax
#RRGGBBAA   // The eight-value syntax
```

Both upper- and lower-case hex digits are supported.

Style writers can use standard hex colors with highlighting and color picker support in the IDE. 

image

# Implementation strategy and design decisions

* `RGBA` is the ground-truth storage format in the Style compiler, and `#rrggbbaa` is the output format used by the color picker.
* Add tests to validate values of hex literals
* Updated `continousmap.sty` to use hex literals
* Side mission: found a bunch of very duplicated code in `Style.test.ts`. Pulled them into `loadProgs` to simplify things.

# Examples with steps to reproduce them

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new ESLint warnings
- [x] I have reviewed any generated changes to the `diagrams/` folder

# Open questions

Questions that require more discussion or to be addressed in future development:

### CODE REVIEW

File: packages/core/src/compiler/Style.test.ts [hunk 1]
```diff
@@ -56,17 +59,65 @@ export const loadProgs = ({
...
+
+  // third pass: compile all expressions in topological sorted order
+  const translation = S.translate(
+    { makeInput },
+    canvas,
+    graph,
+    assignment.diagnostics.warnings
+  );
+
+  const state = S.compileStyle(
```
Comment (wodeni) [inline packages/core/src/compiler/Style.test.ts:109]: I haven't really copied the entirety of `compileStyle` in here so this is technically running things twice. Wondering if there's a better way to do this.

File: packages/core/src/parser/ParserUtil.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export const basicSymbols: moo.Rules = {
   rparen: ")",
   apos: "'",
   comma: ",",
+  hex_literal: /#(?:[0-9A-Fa-f]{3,})/,
```
Comment (samestep) [inline packages/core/src/parser/ParserUtil.ts:20]: Why do you use `(?:)` here?

File: packages/core/src/utils/Util.ts [hunk 3]
```diff
@@ -177,14 +177,75 @@ export const toScreen = (
 
 //#region color
 
-export const toHexRGB = (color: [number, number, number]): string => {
+export const hexToRgba = (
+  hex: string
+): [number, number, number, number] | null => {
```
Comment (samestep) [inline packages/core/src/utils/Util.ts:182]: By our conventions from #873:
```suggestion
): [number, number, number, number] | undefined => {
```

File: packages/core/src/parser/ParserUtil.ts (hunk 2 above)
Comment (wodeni) [reply packages/core/src/parser/ParserUtil.ts:20]: Probably from a previous attempt at capturing just the hex string. Removing now.

File: packages/core/src/compiler/Style.test.ts (hunk 1 above)
Comment (wodeni) [reply packages/core/src/compiler/Style.test.ts:109]: Yeah, I basically just want to run the compiler up to a certain pass. @liangyiliang had some tests that ran it all the way, but many others only need the translation.

**Pull request merged**